### DISCUSSION TITLE
How to use multiple versions of Refit in one project?

### DISCUSSION THREAD

**Discussion by siriak** [reactiveui/refit#1642] [Q&A] [answered]
Let's say I consume 2 APIs: A and B. They both provide a NuGet package with Refit interfaces and DTOs inside. But A uses Refit 5.* and B uses Refit 7.*. When I use both interfaces in the same project, one of them works and another fails with:

> {class name} doesn't look like a Refit interface. Make sure it has at least one method with a Refit HTTP method attribute and Refit is installed in the project.

I assume it must be because attributes from different versions of Refit are considered to be different types at runtime and Refit cannot find the attribute it is looking for.

What can I do to use both interfaces in the same project?

**Comment by codymullins** [comment]
Interested in a good answer here too.

**Comment by glennawatson** [comment, marked as answer]
You cannot load two different major versions of the `Refit` assembly into a single project. The .NET runtime resolves one `Refit.dll` per app, so whichever version wins, the other package's generated clients bind against attribute/runtime types that are not there at runtime. That is exactly the "doesn't look like a Refit interface" error you are seeing.

Refit also generates the client implementation in the assembly that defines the interface (the A or B package), against the Refit version that package was built with. There is no binding-redirect fix because the generated code and the runtime types must match versions.

Practical options:

- Ask both vendors to target the same Refit major version. This is the only clean fix.
- Isolate the mismatched API behind a separate assembly/process (e.g. a small service or a plugin AssemblyLoadContext) that references its own Refit version, and call it across that boundary.
- Drop Refit for one of the two APIs and hand-write a thin `HttpClient` wrapper for it, keeping Refit for the other.

There is no supported way to run Refit 5.x and 7.x side by side in the same compilation.
